### ISSUE TITLE
Exception on RestAction#map/flatmap results in the RestAction to stop (and none of the success/failure consumers to run)

### ISSUE THREAD

**Issue by Tais993** [DV8FromTheWorld/JDA#2042]
[download]: https://bintray.com/dv8fromtheworld/maven/JDA/_latestVersion
[guild]: https://discord.gg/0hMr4ce0tIk3pSjp
[stack overflow]: https://stackoverflow.com/questions/tagged/java

## General Troubleshooting
  
- [x] I have checked for similar issues.
- [x] I have updated to the [latest JDA version][download]. (Using 4.4.0_352 and 5.0.0-alpha.8)
- [x] I have checked the branches or the maintainers' PRs for upcoming bug fixes.

## Bug Report

If an exception occurs within the RestAction#map/flatmap function, an error pops up and the RestAction doesn't finish.
> ERROR RestAction - Encountered error while processing success consumer

### Intentional behaviour?

This could be intentional, RestAction#flatmap(Predicate, Function) exists.
But this does mean the issue persists for RestAction#map, and that was my original issue.

So still, RestAction#map is limited with error handling, and my original issue persists.
Seems I've to change the behaviour so I can use RestAction#flatmap instead, or I'll handle it with null? 

### Expected Behavior

The failure consumer to trigger

### Code Example or Reproduction Steps

```java
        jda.retrieveUserById("257500867568205824")
                .flatMap(user -> {
                    Role role = (Role) user;
                    return user.retrieveProfile();
                }).queue(role -> {
                    System.out.println("success, feeling good");
                }, failure -> {
                    System.out.println("OWH NO! SOMETHING WENT WRONG!");
                });
```

### Code for JDABuilder or DefaultShardManagerBuilder Used

```java
        JDA jda = JDABuilder.create("smirk", GatewayIntent.GUILD_MESSAGES)
                .build()
                .awaitReady();
```

### Exception or Error
```java
[ForkJoinPool.commonPool-worker-1] ERROR RestAction - Encountered error while processing success consumer
java.lang.ClassCastException: class net.dv8tion.jda.internal.entities.UserImpl cannot be cast to class net.dv8tion.jda.api.entities.Role (net.dv8tion.jda.internal.entities.UserImpl and net.dv8tion.jda.api.entities.Role are in unnamed module of loader 'app')
	at nl.tijsbeek.Application.lambda$main$0(Application.java:20)
	at net.dv8tion.jda.internal.requests.restaction.operator.FlatMapRestAction.supply(FlatMapRestAction.java:44)
	at net.dv8tion.jda.internal.requests.restaction.operator.FlatMapRestAction.lambda$queue$0(FlatMapRestAction.java:54)
	at net.dv8tion.jda.api.requests.Request.lambda$onSuccess$0(Request.java:95)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
```

**Comment by MinnDevelopment** [comment]
Can you try if this works as expected with #2044

**Comment by Tais993** [comment]
Looks like this does cause some unwanted behaviour.
When using this as code:
```java
        jda.retrieveUserById("257500867568205824")
                .map(user -> {
                    Role role = (Role) user;
                    return user.retrieveProfile();
                }).queue(role -> {
                    System.out.println("success, feeling good");
                }, failure -> {
                    failure.printStackTrace();
                });
```
It results this as error:
```
java.lang.ClassCastException: class net.dv8tion.jda.internal.entities.UserImpl cannot be cast to class net.dv8tion.jda.api.entities.Role (net.dv8tion.jda.internal.entities.UserImpl and net.dv8tion.jda.api.entities.Role are in unnamed module of loader 'app')
	at nl.tijsbeek.Application.lambda$main$0(Application.java:20)
	at net.dv8tion.jda.internal.requests.restaction.operator.MapRestAction.lambda$queue$0(MapRestAction.java:42)
	at net.dv8tion.jda.internal.requests.restaction.operator.RestActionOperator.lambda$handle$0(RestActionOperator.java:64)
	at net.dv8tion.jda.api.requests.Request.lambda$onSuccess$0(Request.java:95)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: net.dv8tion.jda.api.exceptions.ContextException
	at net.dv8tion.jda.api.exceptions.ContextException.here(ContextException.java:54)
	at net.dv8tion.jda.internal.requests.restaction.operator.RestActionOperator.contextWrap(RestActionOperator.java:124)
	at net.dv8tion.jda.internal.requests.restaction.operator.RestActionOperator.handle(RestActionOperator.java:59)
Caused by: net.dv8tion.jda.api.exceptions.ContextException

	at net.dv8tion.jda.internal.requests.restaction.operator.MapRestAction.queue(MapRestAction.java:41)
	at nl.tijsbeek.Application.main(Application.java:22)
```

Not sure or it's your intention to have an error like this?
These "caused by ContextException" look weird?
Regardless, it does fix the initial issue so thanks.

**Comment by MinnDevelopment** [comment]
> Not sure or it's your intention to have an error like this? (random empty line)

This doesn't seem to be caused by JDA, but rather by whatever is used to print the exception. JDA correctly initializes the exception cause and that logic hasn't changed for a while.

**Comment by Tais993** [comment]
I see, my fault.

In that case, yes it fixes the issue correctly.

**Comment by MinnDevelopment** [comment]
Fixed in [5.0.0-alpha.10](https://github.com/DV8FromTheWorld/JDA/releases/tag/v5.0.0-alpha.10)
